src/main/webapp/script.js

```diff
@@ -20,28 +20,29 @@ function loadPage() {
 
 //** Checks login status and display HTML elements accordingly. */
 async function checkLog() {
-  var loggedIn;
   const response = await fetch("/login");
   const loginInfo = await response.json();
 
-  // Add correspondent link to the log button.
-  const logButton = document.getElementById("log-button");
-  logButton.href = loginInfo.actionUrl;
-
-  // User is logged in.
-  if (loginInfo.loggedIn === true) {
-    document.getElementById("profile").style.visibility = "visible";
-    logButton.innerText = "LOGOUT";
-    document.getElementById("route-content").style.visibility = "visible";
-    loggedIn = true;
-    // User is not logged in.
+  if (loginInfo.success) {
+    // Add correspondent link to the log button.
+    const logButton = document.getElementById("log-button");
+    logButton.href = loginInfo.object.actionUrl;
+
+    // User is logged in.
+    if (loginInfo.object.loggedIn === true) {
+      document.getElementById("profile").style.visibility = "visible";
+      logButton.innerText = "LOGOUT";
+      document.getElementById("route-content").style.visibility = "visible";
+      // User is not logged in.
+    } else {
```

Comment:
I think it's better to put comments like this:
```

if (loginInfo.object.loggedIn === true) {
  // User is logged in.
  ...
} else {
  // User is not logged in.
  ...
```

Reply:
Done